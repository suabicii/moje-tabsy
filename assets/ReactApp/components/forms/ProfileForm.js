import React from "react";

function ProfileForm() {
    return (
        <div className="card mt-3">
            <div className="card-header text-center">Ustawienia profilu</div>
            <div className="card-body">
                <form role="form" name="profile_form" onSubmit={e => {
                    e.preventDefault();
                }}>
                    <div className="form-floating mt-3">
                        <input type="text" className="form-control" id="name" name="name"
                               placeholder="ImiÄ™" data-testid="firstName" required
                        />
                        <label htmlFor="name">ImiÄ™</label>
                    </div>
                    <div className="form-floating mt-3">
                        <input type="email" className="form-control" id="email" name="email"
                               placeholder="Email" data-testid="email" required
                        />
                        <label htmlFor="name">Email</label>
                    </div>
                    <div className="input-group mt-3">
                        <div className="input-group input-group--prefix-number">
                            <span className="input-group-text">+</span>
                            <select id="tel_prefix" name="tel_prefix"
                                    className="form-select" aria-label="Prefix">
                                <option value="48">48 ðŸ‡µðŸ‡±</option>
                                <option value="47">47 ðŸ‡³ðŸ‡´</option>
                            </select>
                        </div>
                        <input type="text" id="tel" name="tel"
                               className="form-control rounded-start" placeholder="Nr tel"/>
                    </div>
                    <div className="form-floating mt-3">
                        <input type="password" className="form-control" id="oldPassword" name="oldPassword"
                               placeholder="Stare hasÅ‚o" data-testid="oldPassword"
                        />
                        <label htmlFor="name">Stare hasÅ‚o</label>
                    </div>
                    <div className="form-floating mt-3">
                        <input type="password" className="form-control" id="newPassword" name="newPassword"
                               placeholder="Nowe hasÅ‚o" data-testid="newPassword"
                        />
                        <label htmlFor="name">Nowe hasÅ‚o</label>
                    </div>
                    <div className="form-floating mt-3">
                        <input type="password" className="form-control" id="newPasswordRepeated"
                               name="newPasswordRepeated"
                               placeholder="Nowe hasÅ‚o" data-testid="newPasswordRepeated"
                        />
                        <label htmlFor="name">PowtÃ³rz nowe hasÅ‚o</label>
                    </div>
                    <div className="d-grid mt-3">
                        <button className="btn btn-primary">Zapisz <i className="fa-solid fa-floppy-disk"></i></button>
                    </div>
                </form>
            </div>
        </div>
    );
}

export default ProfileForm;