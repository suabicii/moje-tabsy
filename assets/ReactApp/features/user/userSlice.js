import {createSlice} from "@reduxjs/toolkit";
import {fetchData} from "../../utils/fetchData";

const userSlice = createSlice({
    name: 'user',
    initialState: {},
    reducers: {
        setUserData: (state, action) => action.payload
    }
});

const {reducer, actions} = userSlice;
export const {setUserData} = actions;

export const fetchUserData = () => async dispatch => {
    await fetchData('/api/user-data').then(data => {
        dispatch(setUserData(data))
    }).catch(err => {
        console.log(err);
    });
};

export default reducer;