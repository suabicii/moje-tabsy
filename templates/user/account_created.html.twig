{% extends 'base.html.twig' %}

{% block title %}{% endblock %}

{% block body %}
    <body>
    <main>
        <section class="user-info position-relative">
            <div class="user-info__overlay position-absolute top-50 start-50 translate-middle text-center bg-white p-4 rounded">
                <h1 class="display-3">DziÄ™kujÄ™ za rejestracjÄ™! ğŸ¤©</h1>
                <p class="lead">Å»eby mÃ³c w peÅ‚ni korzystaÄ‡ z serwisu musisz jeszcze <strong>aktywowaÄ‡</strong> swoje konto.</p>
                <p>Aby to zrobiÄ‡, wystarczy, Å¼e klikniesz link aktywacyjny umieszczony w wiadomoÅ›ci e-mail <i class="fa-solid fa-envelope"></i>.</p>
                <p class="text-info">JeÅ›li nie moÅ¼esz znaleÅºÄ‡ maila aktywacyjnego w swojej skrzynce odbiorczej <i class="fa-solid fa-inbox"></i>, sprawdÅº folder SPAM.</p>
                <div class="grid mt-2">
                    <a href="{{ path('front_page') }}">PowrÃ³t do strony gÅ‚Ã³wnej</a>
                    <a href="{{ path('login_page') }}" class="mt-1 mt-lg-0">Zaloguj siÄ™</a>
                </div>
            </div>
        </section>
    </main>
    {% block javascripts %}
        {{ parent() }}
    {% endblock %}
    </body>
{% endblock %}