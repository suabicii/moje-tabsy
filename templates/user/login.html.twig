{% extends 'base.html.twig' %}

{% block title %}Zaloguj się{% endblock %}

{% block body %}
    <body>
    <main>
        <section class="sign-in position-relative">
            <div class="sign-in__overlay position-absolute top-50 start-50 translate-middle text-center bg-white p-4 rounded">
                <h1>Logowanie</h1>
                <a href="{{ path('front_page') }}">Powrót do strony głównej</a>
                <form action="dashboard.html" class="mt-2">
                    <div class="form-floating mb-3">
                        <input type="email" class="form-control" id="floatingInput" placeholder="name@example.com">
                        <label for="floatingInput">Email</label>
                    </div>
                    <div class="form-floating">
                        <input type="password" class="form-control" id="floatingPassword" placeholder="Hasło">
                        <label for="floatingPassword">Hasło</label>
                    </div>
                    <div class="d-grid">
                        <button type="submit" class="btn btn-primary mt-2">Zaloguj się <i
                                    class="fa-solid fa-arrow-right-to-bracket"></i></button>
                    </div>
                </form>
                <div class="grid">
                    <a href="#" class="mt-2">Nie pamiętam hasła</a>
                    <a href="{{ path('register_page') }}" class="mt-2">Nie masz jeszcze konta? Zarejestruj się!</a>
                </div>
            </div>
        </section>
    </main>
    {% block javascripts %}
        {{ parent() }}
    {% endblock %}
    </body>
{% endblock %}
