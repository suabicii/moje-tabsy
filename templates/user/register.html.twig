{% extends 'base.html.twig' %}

{% block title %}{% endblock %}

{% block body %}
    <body>
    <main>
        <section class="sign-up position-relative">
            <div class="sign-up__overlay position-absolute top-50 start-50 translate-middle text-center bg-white p-4 rounded">
                <h1>Rejestracja</h1>
                <a href="{{ path('front_page') }}">Powrót do strony głównej</a>
                <form class="mt-2">
                    <div class="form-floating mb-3">
                        <input type="email" class="form-control" id="floatingName" placeholder="Imię" required>
                        <label for="floatingName">Imię</label>
                    </div>
                    <div class="form-floating mb-3">
                        <input type="email" class="form-control" id="floatingEmail" placeholder="name@example.com" required>
                        <label for="floatingEmail">Email</label>
                    </div>
                    <div class="form-floating mb-3">
                        <input type="password" class="form-control" id="floatingPassword" placeholder="Hasło" required>
                        <label for="floatingPassword">Hasło</label>
                    </div>
                    <div class="form-floating">
                        <input type="password" class="form-control" id="floatingPasswordRepeat" placeholder="Powtórz hasło" required>
                        <label for="floatingPasswordRepeat">Powtórz hasło</label>
                    </div>
                    <div class="d-grid">
                        <button type="submit" class="btn btn-primary mt-2">Zarejestruj się <i class="fa-solid fa-user-plus"></i>
                        </button>
                    </div>
                </form>
                <div class="grid">
                    <a href="{{ path('login_page') }}" class="mt-2">Masz już konto? Zaloguj się!</a>
                </div>
            </div>
        </section>
    </main>
    {% block javascripts %}
        {{ parent() }}
    {% endblock %}
    </body>
{% endblock %}
