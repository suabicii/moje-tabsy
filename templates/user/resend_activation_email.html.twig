{% extends 'base.html.twig' %}

{% block title %}Wyślij ponownie email aktywacyjny{% endblock %}

{% block body %}
    <body>
    <main>
        <section class="sign-in position-relative">
            <div class="sign-in__overlay position-absolute top-50 start-50 translate-middle text-center bg-white p-4 rounded">
                <h1>Wyślij ponownie wiadomość email z linkiem aktywacyjnym</h1>
                <div class="d-grid">
                    <a href="{{ path('login_page') }}">Wstecz</a>
                    <a href="{{ path('front_page') }}" class="mt-2">Powrót do strony głównej</a>
                </div>
                <form action="{{ path('resend_activation_email_submit') }}" name="resend_activation_email_form"
                      class="mt-3" method="post">
                    <div class="form-floating">
                        <input type="email" class="form-control" id="floatingInput" placeholder="name@example.com"
                               name="email">
                        <label for="floatingInput">Email</label>
                    </div>
                    {% if error %}
                        <div class="alert alert-danger mt-2 mb-1" role="alert">
                            <div>
                                <p class="lead text-white"><i class="fa-solid fa-circle-exclamation"></i> Błędy:</p>
                                <p class="text-white">{{ error }}</p>
                            </div>
                        </div>
                    {% endif %}
                    <div class="d-grid mt-2">
                        <button id="submit" type="submit" class="btn btn-primary mt-2">
                            Wyślij <i class="fa-solid fa-paper-plane"></i></button>
                    </div>
                </form>
            </div>
        </section>
    </main>
    {% block javascripts %}
        {{ parent() }}
    {% endblock %}
    </body>
{% endblock %}