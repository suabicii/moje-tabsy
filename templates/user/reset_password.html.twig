{% extends 'base.html.twig' %}

{% block title %}Nie pamiętam hasła{% endblock %}

{% block body %}
    <body>
    <main>
        <section class="sign-in position-relative">
            <div class="sign-in__overlay position-absolute top-50 start-50 translate-middle text-center bg-white p-4 rounded">
                <h1>Wpisz adres email konta, do którego chcesz odzyskać dostęp</h1>
                <div class="d-grid">
                    <a href="{{ path('login_page') }}">Wstecz</a>
                    <a href="{{ path('front_page') }}" class="mt-2">Powrót do strony głównej</a>
                </div>
                <form action="{{ path('reset_password_submit') }}" name="reset_password_form"
                      class="mt-3" method="post">
                    <div class="form-floating">
                        <input type="email" class="form-control" id="floatingInput" placeholder="name@example.com"
                               name="email">
                        <label for="floatingInput">Email</label>
                    </div>
                    {% if error %}
                        <div class="alert alert-danger mt-2 mb-1" role="alert">
                            <div>
                                <p class="lead text-white"><i class="fa-solid fa-circle-exclamation"></i> Błędy:</p>
                                <p class="text-white">{{ error }}</p>
                            </div>
                        </div>
                    {% endif %}
                    <div class="d-grid mt-2">
                        <button id="submit" type="submit" class="btn btn-primary mt-2">
                            Resetuj hasło <i class="fa-solid fa-paper-plane"></i></button>
                    </div>
                </form>
            </div>
        </section>
    </main>
    {% block javascripts %}
        {{ parent() }}
    {% endblock %}
    </body>
{% endblock %}