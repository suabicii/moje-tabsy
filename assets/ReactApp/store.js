import {configureStore} from "@reduxjs/toolkit";
import drugsSlice from "./features/drugs/drugsSlice"
import userSlice from "./features/user/userSlice";
import darkModeSlice from "./features/darkMode/darkModeSlice";

export default configureStore({
    reducer: {
        drugs: drugsSlice,
        user: userSlice,
        darkMode: darkModeSlice
    }
});