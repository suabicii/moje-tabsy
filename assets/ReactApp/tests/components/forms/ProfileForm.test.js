/**
 * @jest-environment jsdom
 * */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import ProfileForm from "../../../components/forms/ProfileForm";
import {act, fireEvent, render, screen, waitFor} from "@testing-library/react";
import {userData} from '../fixtures/apiData';
import {Provider} from "react-redux";
import store from "../../../store";
import {fetchUserData} from "../../../features/user/userSlice";

const setMockedFetch = (mockedResponse, isFormSubmit = false) => {
    jest.spyOn(global, 'fetch').mockImplementation(() => Promise.resolve({
        json: () => Promise.resolve(mockedResponse)
    }));

    if (!isFormSubmit) {
        store.dispatch(fetchUserData());
    }
};

const renderProfileForm = () => {
    render(
        <Provider store={store}>
            <ProfileForm/>
        </Provider>
    );
};

afterAll(() => {
    global.fetch.mockClear();
    delete global.fetch;
});

it('should correctly display ProfileForm', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
        <Provider store={store}>
            <ProfileForm/>
        </Provider>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('should display ProfileForm with user data fetched from API', async () => {
    setMockedFetch(userData);

    renderProfileForm();

    await waitFor(() => expect(screen.getByRole('form')).toHaveFormValues({
        name: userData.name,
        email: userData.email,
        tel: userData.tel
    }));
});

it('should display modal with positive information after form submission if status is OK', async () => {
    setMockedFetch(userData);
    renderProfileForm();

    setMockedFetch({status: 200}, true);
    await act(() => {
        fireEvent.change(screen.getByTestId('email'), {target: {value: 'new@email.com'}});
        fireEvent.submit(screen.getByRole('form'));
    });

    expect(screen.getByTestId('status-ok')).toBeVisible();
});

it('should display modal with negative information after form submission if status is FAIL', async () => {
    setMockedFetch(userData);
    renderProfileForm();

    await act(() => {
        fireEvent.change(screen.getByTestId('firstName'), {target: {value: 'John'}});
        fireEvent.submit(screen.getByRole('form'));
    });

    expect(screen.getByTestId('status-failed')).toBeVisible();
});