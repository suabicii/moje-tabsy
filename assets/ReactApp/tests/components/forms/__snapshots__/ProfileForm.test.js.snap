// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly display ProfileForm 1`] = `
<div
  className="card mt-3"
>
  <div
    className="card-header text-center"
  >
    Ustawienia profilu
  </div>
  <div
    className="card-body"
  >
    <form
      name="profile_form"
      onSubmit={[Function]}
      role="form"
    >
      <div
        className="form-floating mt-3"
      >
        <input
          className="form-control"
          data-testid="firstName"
          id="name"
          name="name"
          placeholder="ImiÄ™"
          required={true}
          type="text"
        />
        <label
          htmlFor="name"
        >
          ImiÄ™
        </label>
      </div>
      <div
        className="form-floating mt-3"
      >
        <input
          className="form-control"
          data-testid="email"
          id="email"
          name="email"
          placeholder="Email"
          required={true}
          type="email"
        />
        <label
          htmlFor="name"
        >
          Email
        </label>
      </div>
      <div
        className="input-group mt-3"
      >
        <div
          className="input-group input-group--prefix-number"
        >
          <span
            className="input-group-text"
          >
            +
          </span>
          <select
            aria-label="Prefix"
            className="form-select"
            id="tel_prefix"
            name="tel_prefix"
          >
            <option
              value="48"
            >
              48 ðŸ‡µðŸ‡±
            </option>
            <option
              value="47"
            >
              47 ðŸ‡³ðŸ‡´
            </option>
          </select>
        </div>
        <input
          className="form-control rounded-start"
          id="tel"
          name="tel"
          placeholder="Nr tel"
          type="text"
        />
      </div>
      <div
        className="form-floating mt-3"
      >
        <input
          className="form-control"
          data-testid="oldPassword"
          id="oldPassword"
          name="oldPassword"
          placeholder="Stare hasÅ‚o"
          type="password"
        />
        <label
          htmlFor="name"
        >
          Stare hasÅ‚o
        </label>
      </div>
      <div
        className="form-floating mt-3"
      >
        <input
          className="form-control"
          data-testid="newPassword"
          id="newPassword"
          name="newPassword"
          placeholder="Nowe hasÅ‚o"
          type="password"
        />
        <label
          htmlFor="name"
        >
          Nowe hasÅ‚o
        </label>
      </div>
      <div
        className="form-floating mt-3"
      >
        <input
          className="form-control"
          data-testid="newPasswordRepeated"
          id="newPasswordRepeated"
          name="newPasswordRepeated"
          placeholder="Nowe hasÅ‚o"
          type="password"
        />
        <label
          htmlFor="name"
        >
          PowtÃ³rz nowe hasÅ‚o
        </label>
      </div>
      <div
        className="d-grid mt-3"
      >
        <button
          className="btn btn-primary"
        >
          Zapisz 
          <i
            className="fa-solid fa-floppy-disk"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
